java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -744,7 +744,7 @@ public void setValueCount(int valueCount) {
     }
     /* valueCount for the data vector is the current end offset */
     final int childValueCount = (valueCount == 0) ? 0 :
-            offsetBuffer.getInt(valueCount * OFFSET_WIDTH);
+            offsetBuffer.getInt(lastSet * OFFSET_WIDTH);
```

Comment:
So were we always setting the value count for child vector incorrectly? I am not clear what the problem is. Is there a special case we have run into?

Reply:
Hi!  I ran into this over on #3294, when I tried to convert a string array from a JDBC field into a VarChar ListVector.  If one of the values in the JDBC string array was `null`, I couldn't read the value after it from the VarChar ListVector (the following value came back as an empty string).

@praveenbingo was kind enough to put together this PR to fix the issue.

Reply:
Yup, @siddharthteotia - we were not correctly handling lists that could have null values (setting child vector count using valuecount will not handle cases where valuecount < lastSet i.e. presence of nulls).

It does not affect List of FixedWidthVectors and only VariableWidthVectors. 

Please let me know if you would need more information.

Reply:
Thanks @mikepigott  and @praveenbingo